systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java

```diff
@@ -129,6 +129,59 @@ void testReceiveSimpleMessageTlsScramSha() throws Exception {
         assertThat(deleteConsumer(bridgeHost, bridgePort, groupId, name), is(true));
     }
 
+    @Test
+    void testScramShaAuthWithWeirdNamedUser() throws Exception {
+        String topicName = "topic" + rng.nextInt(Integer.MAX_VALUE);
+        String groupId = "my-group-" + rng.nextInt(Integer.MAX_VALUE);
+
+        // Create weird named user with . and more than 64 chars -> SCRAM-SHA
+        String weirdUserName = "jjglmahyijoambryleyxjjglmahy.ijoambryleyxjjglmahyijoambryleyxasd.asdasidioiqweioqiweooioqieioqieoqieooi";
+
+        // Create topic
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName).done();
+        // Create user
+        KafkaUserResource.scramShaUser(CLUSTER_NAME, weirdUserName).done();
+
+        JsonObject config = new JsonObject();
+        config.put("name", weirdUserName);
+        config.put("format", "json");
+        config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        // Create consumer
+        JsonObject response = createBridgeConsumer(config, bridgeHost, bridgePort, groupId);
+        assertThat("Consumer wasn't created correctly", response.getString("instance_id"), is(weirdUserName));
+
+        // Create topics json
+        JsonArray topic = new JsonArray();
+        topic.add(topicName);
+        JsonObject topics = new JsonObject();
+        topics.put("topics", topic);
+
+        // Subscribe
+        assertThat(HttpUtils.subscribeHttpConsumer(topics, bridgeHost, bridgePort, groupId, weirdUserName, client), is(true));
+
+        BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+                .withTopicName(topicName)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(CLUSTER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withSecurityProtocol(SecurityProtocol.SASL_SSL)
+                .withKafkaUsername(weirdUserName)
+                .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+                .build();
```

Comment:
`indent` in InteliJ set preferences -> editor -> code style -> java -> (set everything to 4 |tab size, indent, continuationn indent) to aviod all these 8's idents.

Reply:
thanks for tip :)